org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/ScalaSourceFileEditor.scala

```diff
@@ -439,7 +440,35 @@ class ScalaSourceFileEditor
       else
         super.requestWidgetToken(requester, priority)
   }
+  private var outlinePage:ScalaOutlinePage = null
```

Comment:
The outline page also needs to work for `ScalaClassFileEditor`. Please move this to `ScalaCompilationUnitEditor` to handle both.